Detectors/TPC/spacecharge/src/NearestNeighbour.cxx

```diff
@@ -0,0 +1,78 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file  NearestNeighbour.cxx
+/// \brief Definition of NearestNeighbour class
+///
+/// \author  Matthias Kleiner <mkleiner@ikf.uni-frankfurt.de>
+/// \date Nov 23, 2020
+
+#include "TPCSpaceCharge/NearestNeighbour.h"
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Search_traits_3.h>
+#include <CGAL/Search_traits_adapter.h>
+#include <CGAL/Orthogonal_k_neighbor_search.h>
+#include <CGAL/property_map.h>
+#include <boost/iterator/zip_iterator.hpp>
+#include <vector>
+
+using Kernel = CGAL::Exact_predicates_inexact_constructions_kernel;
+using Point_3 = Kernel::Point_3;
+using Point_Index = boost::tuple<Point_3, std::array<unsigned int, 3>>;
+using Traits_base = CGAL::Search_traits_3<Kernel>;
+using Traits = CGAL::Search_traits_adapter<Point_Index, CGAL::Nth_of_tuple_property_map<0, Point_Index>, Traits_base>;
+using K_neighbor_search = CGAL::Orthogonal_k_neighbor_search<Traits>;
+using Tree = K_neighbor_search::Tree;
+
+std::unique_ptr<std::vector<Point_3>> mPoints;
```

Comment:
You can forward declare a class NearestNeighbourInternal in NearestNeighbour.h and add a member std::unique_ptr<NearestNeighbourInternal> to the NearestNeighbour class.
Then, define the actual ...Internal class inside NearestNeightbour.cxx, and move these global objects in there and create the internal object in the constructor.
Only thing you have to make sure is that both constructor and destructor must be in the .cxx and not in the header. For the destructor, you can declare it in the header and add to the .cxx `NearestNeighbour::~NearestNeighbour() = default;`.

Reply:
Thanks for the nice explanation :) I moved it to an internal class as suggested